### PR TITLE
deps(dependencies): Address high-severity vulnerabilities by updating dependencies

### PR DESCRIPTION
**Pull request by keenwilson**
## Problem

The aws-toolkit-vscode repository has several high-severity vulnerabilities in its dependencies, as identified by the npm audit command. These vulnerabilities, if left unaddressed, could pose a security risk to users of the AWS Toolkit for VS Code extension.

The vulnerabilities include:

- elliptic package versions 2.0.0 to 6.5.6 have multiple vulnerabilities related to EDDSA, ECDSA, and BER-encoded signatures.
- es5-ext package versions 0.10.1 to 0.10.62 are vulnerable to Regular Expression Denial of Service (ReDOS) in the function#copy and function#toStringTokens functions.
- fast-xml-parser package versions prior to 4.4.1 are vulnerable to ReDOS at currency parsing.

## Solution

To address these high-severity vulnerabilities, this pull request updates the dependencies in the aws-toolkit-vscode repository:

- The elliptic package is upgraded from version 6.5.3 to 6.5.4 to address the EDDSA, ECDSA, and BER-encoded signatures vulnerabilities.
- The es5-ext package is upgraded from version 0.10.53 to 0.10.62 to address the ReDOS vulnerability in the function#copy and function#toStringTokens functions.
- The fast-xml-parser package is upgraded from version 3.19.0 to 4.4.1 to address the ReDOS vulnerability at currency parsing.

## Implementation Details:

By updating the following top level dependencies, the underlying transitive dependencies with the vulnerabilities were updated to a safe version:

- `@aws-sdk/client-cognito-identity`
- `@aws-sdk/client-lambda`

---

License: I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by nkomonen-amazon** [comment]
Tested all Authentication scenarios for both Amazon Q and AWS Toolkit extensions and ensured they still work. Overall LGTM.

**Comment by AwsBrettBailey** [comment]
@nkomonen-amazon replying for @keenwilson but, yes, we are fine to wait to merge.

**Comment by keenwilson** [comment]
> Tested all Authentication scenarios for both Amazon Q and AWS Toolkit extensions and ensured they still work. Overall LGTM.
> 
> @keenwilson are we fine to merge this after Thursday release so that it has a week to bake?

That works for us. Thank you for your help getting this security fix it.

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+## [Unreleased]
```
Comment (justinmk3) [inline CHANGELOG.md:1]: 1. See [changelog guidelines](https://github.com/aws/aws-toolkit-vscode/blob/master/CONTRIBUTING.md#changelog).
2. To add a new changelog entry: `npm run newChange -w packages/amazonq` or `npm run newChange -w packages/toolkit`

File: packages/core/package.json [hunk 2]
```diff
@@ -4101,8 +4101,8 @@
     },
     "dependencies": {
         "@amzn/codewhisperer-streaming": "file:../../src.gen/@amzn/codewhisperer-streaming",
-        "@aws-sdk/client-cognito-identity": "3.490.0",
-        "@aws-sdk/client-lambda": "3.385.0",
+        "@aws-sdk/client-cognito-identity": "^3.637.0",
+        "@aws-sdk/client-lambda": "^3.637.0",
         "@aws-sdk/client-sso": "^3.342.0",
         "@aws-sdk/client-sso-oidc": "^3.574.0",
         "@aws-sdk/credential-provider-ini": "3.46.0",
```
Comment (justinmk3) [inline packages/core/package.json:4108]: These sdk v3 packages are drifting very far from each other. That's probably why package-lock.json now has 6k lines of workarounds

**Pull request merged**